lib/jekyll/document.rb

```diff
@@ -404,28 +418,29 @@ def respond_to_missing?(method, *)
     #
     # Returns nothing.
     def categories_from_path(special_dir)
-      superdirs = relative_path.sub(%r!#{special_dir}(.*)!, "")
-        .split(File::SEPARATOR)
-        .reject do |c|
-        c.empty? || c == special_dir || c == basename
-      end
+      superdirs = relative_path.sub(Document.superdirs_regex(special_dir), "")
+      superdirs = superdirs.split(File::SEPARATOR)
+      superdirs.reject! { |c| c.empty? || c == special_dir || c == basename }
```

Comment:
It doesn't look like you need all three assignments for this.
```ruby
 superdirs = relative_path.sub(%r!#{special_dir}(.*)!, "").
        split(File::SEPARATOR).
        reject! { |c| c.empty? || c == special_dir || c == basename }
```

Whether or not you want to change it, I'll leave up to you. Just pointing it out. 😄

Reply:
There's a small catch though. I can't chain `:reject!` to the end because it will return `nil` in some cases.. :smiley: